bitcoin/base58.c

```diff
@@ -53,16 +51,22 @@ static bool from_base58(u8 *version,
 {
 	u8 buf[1 + sizeof(*rmd) + 4];
 	/* Avoid memcheck complaining if decoding resulted in a short value */
-	memset(buf, 0, sizeof(buf));
-	b58_sha256_impl = my_sha256;
-
 	size_t buflen = sizeof(buf);
-	b58tobin(buf, &buflen, base58, base58_len);
-
-	int r = b58check(buf, buflen, base58, base58_len);
+	memset(buf, 0, buflen);
+	char *terminated_base58 = malloc(base58_len + 1);
+	memcpy(terminated_base58, base58, base58_len);
+	terminated_base58[base58_len] = '\0';
```

Comment:
This might be a candidate for `tal_arr_dup(NULL, char, base58_len, 1)` which allocates `base58_len + 1` and copies. You still need to set the terminating `\0`.

Reply:
done